plugins/out_s3/s3.c

```diff
@@ -50,6 +51,19 @@ static struct multipart_upload *get_upload(struct flb_s3 *ctx,
 static struct multipart_upload *create_upload(struct flb_s3 *ctx,
                                               const char *tag, int tag_len);
 
+static struct flb_aws_header content_encoding_header = {
+    .key = "Content-Encoding",
```

Comment:
More on my comment below about concurrency. This header is safe to use directly in concurrent requests because its read-only. You won't be changing any of its fields. 

The `content_type_header` below is not concurrent safe since you write to it